### PR TITLE
fix: json scalar types rendering not supported for graphql api

### PR DESCRIPTION
**Pull request by spetrunin**
depends on: https://github.com/wundergraph/graphql-go-tools/pull/481
fixes: #448 

#### Checklist

- [ ] run `make test`
- [ ] PR title must follow [conventional-commit-standard](https://github.com/wundergraph/wundergraph/blob/main/CONTRIBUTING.md#conventional-commit-standard)
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Code of conduct](https://github.com/wundergraph/wundergraph/blob/main/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: testapps/simple/README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# WunderGraph simple example
+
+#### Getting started
+
+```shell
+npm install && npm start
```
Comment (StarpTech) [inline testapps/simple/README.md:6]: In our workspace, we use pnpm

File: testapps/simple/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
+{
+  "compilerOptions": {
```
Comment (StarpTech) [inline testapps/simple/tsconfig.json:2]: Inherit from the shared tsconfig at the root.

File: testapps/simple/.wundergraph/wundergraph.operations.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import { configureWunderGraphOperations } from '@wundergraph/sdk';
+import type { OperationsConfiguration } from './generated/wundergraph.operations';
+
+export default configureWunderGraphOperations<OperationsConfiguration>({
```
Comment (StarpTech) [inline testapps/simple/.wundergraph/wundergraph.operations.ts:4]: Is that everything needed to be a simple example?

File: testapps/simple/.wundergraph/wundergraph.config.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+						// change this before deploying to production to the actual domain where you're deploying your app
+						'http://localhost:3000',
+				  ]
+				: ['http://localhost:3000', new EnvironmentVariable('WG_ALLOWED_ORIGIN')],
+	},
+	dotGraphQLConfig: {
+		hasDotWunderGraphDirectory: false,
+	},
+	security: {
+		enableGraphQLEndpoint: process.env.NODE_ENV !== 'production' || process.env.GITPOD_WORKSPACE_ID !== undefined,
```
Comment (StarpTech) [inline testapps/simple/.wundergraph/wundergraph.config.ts:34]: GITPOD_WORKSPACE_ID not needed

File: testapps/simple/.wundergraph/wundergraph.config.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
+import { configureWunderGraphApplication, cors, EnvironmentVariable, introspect, templates } from '@wundergraph/sdk';
```
Comment (StarpTech) [inline testapps/simple/.wundergraph/wundergraph.config.ts:1]: Why do we need another testapp?

**Review by StarpTech — CHANGES_REQUESTED** [review]
Looks great, I only made a couple of suggestions for the testapp.

File: testapps/simple/.wundergraph/wundergraph.config.ts (hunk 5 above)
Comment (spetrunin) [reply testapps/simple/.wundergraph/wundergraph.config.ts:1]: yep, it is basically empty starter in context of the workspace

maybe better name will be `debug` or smth like that

it is purpose to been able to debug some issues inside sdk without any additional api

default test app is too huge, contains a lot of features from wundergraph

File: testapps/simple/.wundergraph/wundergraph.config.ts (hunk 4 above)
Comment (spetrunin) [reply testapps/simple/.wundergraph/wundergraph.config.ts]: removed

File: testapps/simple/README.md (hunk 1 above)
Comment (spetrunin) [reply testapps/simple/README.md]: removed readme

File: testapps/simple/tsconfig.json (hunk 2 above)
Comment (StarpTech) [reply testapps/simple/tsconfig.json:2]: ok, its fine :+1:

File: testapps/simple/.wundergraph/wundergraph.config.ts (hunk 5 above)
Comment (StarpTech) [reply testapps/simple/.wundergraph/wundergraph.config.ts:1]: ok, but we should have a structure for which testapps are used for what cases.

Comment (spetrunin) [reply testapps/simple/.wundergraph/wundergraph.config.ts:1]: I have added a draft readme

Comment (StarpTech) [reply testapps/simple/.wundergraph/wundergraph.config.ts:1]: Thank you

**Review by StarpTech — APPROVED** [review]
LGTM

File: packages/sdk/src/graphql/configuration.ts [hunk 6]
```diff
@@ -315,3 +333,53 @@ const addRequiredField = (
...
+
+const findField = (fields: FieldConfiguration[], typeName: string, fieldName: string) => {
+	return fields.find((f) => f.typeName === typeName && f.fieldName === fieldName);
+};
+
+const isJsonField = (type: TypeNode, jsonScalars: string[]) => {
+	const namedTypeName = resolveNamedTypeName(type);
+
+	const scalars = [DefaultJsonType];
+	scalars.push(...jsonScalars);
```
Comment (jensneuse) [inline packages/sdk/src/graphql/configuration.ts:365]: does it make sense to do this for each field? could it be done e.g. at the top of visitSchema?

**Review by jensneuse — CHANGES_REQUESTED** [review]

File: packages/sdk/src/graphql/configuration.ts (hunk 6 above)
Comment (spetrunin) [reply packages/sdk/src/graphql/configuration.ts]: yep, updated

**Review by jensneuse — APPROVED** [review]

**Comment by acdn-sglanzer** [comment]
Would it be possible to publish a new release including this?

**Comment by spetrunin** [comment]
`@wundergraph/sdk@0.131.0` was released with this fix

**Comment by greemo** [comment]
@spetrunin  as commented on the original issue #448 , this fix breaks the logic for returning json strings

**Pull request merged**